### DISCUSSION TITLE
Generating screenshots/images from dashboards

### DISCUSSION THREAD

**Discussion by HerbergSystems** [apache/superset#12525] [General]
Hi,
There is a great function "Download as image" for dashboards on the page. However, we would like to generate an image of a dashboard programmatically on the server. Is there a solution to get an image file or PDF from a dashboard?
Thanks a lot,
kind regards
Jan

**Comment by etr2460** [comment]
Try taking a look at the email report code, it runs a headless browser on a server and takes a screenshot of the dashboard. You may be able to use the email reporting functionality or pull code from it to fulfill your needs

**Comment by mistercrunch** [comment]
@dpgaspar is there an endpoint to get a rendered image of a dashboard? 

Knowing that we have all the mechanics in place to serve report/alerts, it shouldn't be too hard to get an endpoint that serve images if we don't have one already

**Reply by lilila** [reply]
This would be nice!

**Reply by dpgaspar** [reply]
We currently have the following REST API endpoints:

/chart/{pk}/screenshot/{digest}/
and
/dashboard/{pk}/thumbnail/{digest}/

**Comment by HerbergSystems** [comment]
Greetings, Thanks a lot. I will investigate...
Jan

Am 15.01.2021 um 18:52 schrieb Erik Ritter:
> Try taking a look at the email report code, it runs a headless browser
> on a server and takes a screenshot of the dashboard. You may be able to
> use the email reporting functionality or pull code from it to fulfill
> your needs
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/apache/superset/discussions/12525#discussioncomment-285111>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AR4AQF3HWYZC4T6FUUGAQ4LS2B6EVANCNFSM4WCVKWFA>.
> 


-- 
Herberg Systems GmbH
Mr. Jan Herberg
Geschäftsführer/Managing Director
Boschstrasse 23 b
22761 Hamburg - Germany

Phone: +49 40 40 11 346-0
Mobile: +49 173 8683 315
Web: http://www.herberg-systems.com
Amtsgericht Hamburg Registernummer: HR B 77 140
